File: Programming Languages/c++ language/9 object oriented programming/3. Inheritance/Library_Management_System.cpp
#include <bits/stdc++.h>

using namespace std;

class Library{
    public:
    string books[100];
    int noOfBooks = 0;
    Library(){
        string defaultBooks[] = {"KGF", "RRR", "I am back", "Advanced Java", "Python", "You & Me", "Let's code", "OOPs", "King", "The Jungle Books"};
        for(int i = 0; i < 10; i++){
            books[i] = defaultBooks[i];
            noOfBooks++;
        }
    }
};

class Book : public Library{
    public:
    void addBook(string bookName){
        this->books[noOfBooks] = bookName;
        noOfBooks++;
        cout<<"The book "<<bookName<<" added successfully!"<<endl;
    }

    void showBooks(){
        if(noOfBooks == 0){
            cout<<"\nSorry, No any book available!\n"<<endl;
        }
        else{
            cout<<"\n\n_____ Available Books _____\n\n";
            int n = 1;
            for (int i = 0; i < noOfBooks; i++)
            {
                if(books[i] == ""){
                    continue;
                }
                cout<<n<<". "<<books[i]<<endl;
                n++;
            }
            cout<<endl;
        }
    }

    void issueBook(string bookName, string studentName){
        for (int i = 0; i < noOfBooks; i++)
        {
            if (this->books[i] == "")
            {
                continue;
            }
            if(this->books[i] == bookName){
                cout<<"The book "<<bookName<<" has been issued to "<<studentName<<" successfully!"<<endl;
                this->books[i] = "";
                return;
            }
        }
        cout << "Sorry, book " << bookName << " is currently not available!" << endl;
    }

    void returnBook(string bookName, string studentName){
        this->books[noOfBooks] = bookName;
        noOfBooks++;
        cout<<"The book "+bookName<<" has been returned by "<<studentName<<endl;
    }
};


int main(){
    Book b1;
    b1.showBooks();
    b1.addBook("The power of Bhagvat Geeta");
    b1.addBook("Ruby");
    b1.addBook("basic c language");
    b1.addBook("The big bank theory");
    b1.addBook("The kashmir files");
    b1.showBooks();

    b1.issueBook("Python", "Abhishek");
    b1.issueBook("Ruby", "Hariom");
    b1.showBooks();

    b1.returnBook("Ruby", "Hariom");
    b1.showBooks();

    return 0;
}

File: Programming Languages/Java/8. OOPs/Exercises/LibraryManagementSystem.java
import java.util.Scanner;
class Library{
    public String[] books;
    public int noOfBooks;
    Scanner sc = new Scanner(System.in);

    Library(){
        this.books = new String[100];
        this.noOfBooks = 0;

        String[] defaultBooks = {"Think and grow", "The machine learning", "My java", "The python", "Advance C++", "Object Oriented Programming", "My life, My way", "Business analysis", "Social networking", "Health is Wealth", "Kotlin"};

        for (String element: defaultBooks){
            this.books[noOfBooks] = element;
            noOfBooks++;
        }
    }

    void addBook(){
        String book;
        System.out.print("\nEnter new book name : ");
        book = sc.nextLine();
        this.books[noOfBooks] = book;
        noOfBooks++;
        System.out.println("The book \""+book+"\" has been added successfully!");
    }

    void showAvailableBooks(){
        int n = 1;
        System.out.println("\n<<<<< The Available books are >>>>>\n");
        for (String book:this.books){
            if (book == null){
                continue;
            }
            System.out.println(n+". "+book);
            n++;
        }
    }

    void issueBook(){
        String bookName;
        System.out.print("\nEnter book name : ");
        bookName = sc.nextLine();

        for (int i = 0; i<this.books.length; i++){
            if (this.books[i]==null){
                continue;
            }
            if (this.books[i].equals(bookName)){
                System.out.println("The book \""+bookName+"\" has been issued!");
                this.books[i] = null;
                return;
            }
        }
        System.out.println("Sorry, this book \""+bookName+"\" is currently not available!");
    }

    void returnBook(){
        String book;
        System.out.print("\nEnter book name : ");
        book = sc.nextLine();
        this.books[noOfBooks] = book;
        System.out.println("The book \""+book+"\" has been returned!");
        noOfBooks++;
    }
}

public class LibraryManagementSystem {
    public static void main(String[] args) {
        Library theMiddleLibrary = new Library();
        Scanner sc = new Scanner(System.in);

        while (true){
            System.out.println("\nEnter 1 to Show all available books");
            System.out.println("Enter 2 to Issue a book");
            System.out.println("Enter 3 to Return a book");
            System.out.println("Enter 4 to Add a new book");
            System.out.println("Enter 5 to Exit");
            System.out.print("Choice : ");
            int choice = sc.nextInt();

            if (choice == 1){
                theMiddleLibrary.showAvailableBooks();
            }
            else if(choice == 2){
                theMiddleLibrary.issueBook();
            }
            else if(choice == 3){
                theMiddleLibrary.returnBook();
            }
            else if(choice == 4){
                theMiddleLibrary.addBook();
            }
            else if(choice == 5){
                System.out.println("Exiting...");
                break;
            }
            else {
                System.out.println("Invalid choice");
            }
        }
    }
}
